src/coreclr/pal/src/misc/sysinfo.cpp

```diff
@@ -249,426 +237,3 @@ GetSystemInfo(
-        lpBuffer->ullAvailPageFile *= info.mem_unit;
-#endif // HAVE_SYSINFO_WITH_MEM_UNIT
-    }
-#endif // HAVE_SYSINFO
-
-    // Get the physical memory in use - from it, we can get the physical memory available.
-    // We do this only when we have the total physical memory available.
-    if (lpBuffer->ullTotalPhys > 0)
-    {
-#ifndef __APPLE__
-        static volatile bool tryReadMemInfo = true;
-
-        if (tryReadMemInfo)
-        {
-            // Ensure that we don't try to read the /proc/meminfo in successive calls to the GlobalMemoryStatusEx
-            // if we have failed to access the file or the file didn't contain the MemAvailable value.
-            tryReadMemInfo = ReadMemAvailable((uint64_t*)&lpBuffer->ullAvailPhys);
-        }
-
-        if (!tryReadMemInfo)
-        {
-            // The /proc/meminfo doesn't exist or it doesn't contain the MemAvailable row or the format of the row is invalid
-            // Fall back to getting the available pages using sysconf.
-            lpBuffer->ullAvailPhys = sysconf(SYSCONF_PAGES) * sysconf(_SC_PAGE_SIZE);
-        }
-
-        INT64 used_memory = lpBuffer->ullTotalPhys - lpBuffer->ullAvailPhys;
-        lpBuffer->dwMemoryLoad = (DWORD)((used_memory * 100) / lpBuffer->ullTotalPhys);
-#else
-        vm_size_t page_size;
-        mach_port_t mach_port;
-        mach_msg_type_number_t count;
-        vm_statistics_data_t vm_stats;
-        mach_port = mach_host_self();
-        count = sizeof(vm_stats) / sizeof(natural_t);
-        if (KERN_SUCCESS == host_page_size(mach_port, &page_size))
-        {
-            if (KERN_SUCCESS == host_statistics(mach_port, HOST_VM_INFO, (host_info_t)&vm_stats, &count))
-            {
-                lpBuffer->ullAvailPhys = (int64_t)vm_stats.free_count * (int64_t)page_size;
-                INT64 used_memory = ((INT64)vm_stats.active_count + (INT64)vm_stats.inactive_count + (INT64)vm_stats.wire_count) *  (INT64)page_size;
-                lpBuffer->dwMemoryLoad = (DWORD)((used_memory * 100) / lpBuffer->ullTotalPhys);
-            }
-        }
-        mach_port_deallocate(mach_task_self(), mach_port);
-#endif // __APPLE__
-    }
-
-#ifndef TARGET_RISCV64
-    // There is no API to get the total virtual address space size on
-    // Unix, so we use a constant value representing 128TB, which is
-    // the approximate size of total user virtual address space on
-    // the currently supported Unix systems.
-    static const UINT64 VMSize = (1ull << 47);
-#else // TARGET_RISCV64
-    // For RISC-V Linux Kernel SV39 virtual memory limit is 256gb.
-    static const UINT64 VMSize = (1ull << 38);
-#endif // TARGET_RISCV64
-    lpBuffer->ullTotalVirtual = VMSize;
-    lpBuffer->ullAvailVirtual = lpBuffer->ullAvailPhys;
-
-    LOGEXIT("GlobalMemoryStatusEx returns %d\n", fRetVal);
-    PERF_EXIT(GlobalMemoryStatusEx);
-
-    return fRetVal;
-}
-
-bool
-PAL_ReadMemoryValueFromFile(const char* filename, uint64_t* val)
```

Comment:
I think this is used in a few other places.

Reply:
It is used in cgroup, which also is now-unused code in PAL (GC has a separate copy). I think the overarching idea was to move all memory related PAL code to GC, which happened a couple of months ago, but these leftovers weren't cleaned up for some reason at the same time.

Reply:
> cgroup is used to read processor quota limits outside the GC.

Confirmed. I just tried to remove it also and hit that :(

Reply:
Ah, I see. Maybe in the future, we could rewire it so those values are read from single place (GC PAL).